src/main/java/htsjdk/samtools/BamIndexValidator.java

```diff
@@ -100,50 +124,79 @@ public static int exhaustivelyTestIndex(final SamReader reader) { // throws Exce
-            final int numRefs = existingIndex.getNumberOfReferences();
-
-            int chunkCount = 0;
-            int indexCount = 0;
-            for (int i = 0; i < numRefs; i++) {
-
-                final BAMIndexContent content = existingIndex.getQueryResults(i);
-
-                final List<Chunk> chunks = content.getAllChunks();
-                final int numChunks = chunks.size();
-                // We are looking only at the first and last chunks
-                for (final int chunkNo : Arrays.asList(0, numChunks - 1)) {
-                    chunkCount++;
-
-                    final Chunk c = chunks.get(chunkNo);
-                    final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(c));
-                    try {
-                        final SAMRecord sam = iter.next();
-                        iter.close();
-                    } catch (final Exception e) {
-                        throw new SAMException("Exception querying chunk " + chunkNo + " from reference index " + i, e);
+                        final Chunk c = chunks.get(chunkNo);
+                        final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(c));
+                        try {
+                            final SAMRecord sam = iter.next();
+                            iter.close();
+                        } catch (final Exception e) {
+                            throw new SAMException("Exception querying chunk " + chunkNo + " from reference index " + i, e);
+                        }
+                    }
+
+                    // also seek to first and last position in the linear index
+                    final long linearIndexEntries[] = content.getLinearIndex().getIndexEntries();
+                    for (final int binNo : Arrays.asList(0, linearIndexEntries.length - 1)) {
+                        indexCount++;
+                        final long l = linearIndexEntries[binNo];
+                        try {
+                            if (l != 0) {
+                                final CloseableIterator<SAMRecord> iter = ((SamReader.PrimitiveSamReaderToSamReaderAdapter) reader).iterator(new BAMFileSpan(new Chunk(l, l + 1)));
+                                final SAMRecord sam = iter.next();   // read the first record identified by the linear index
+                                iter.close();
+                            }
+                        } catch (final Exception e) {
+                            throw new SAMException("Exception in BamIndexValidator. Linear index access failure " + l + " indexCount=" + indexCount, e);
+                        }
                     }
                 }
+                return chunkCount;
+            } else if (SamIndexes.CSI.fileNameSuffix.endsWith(reader.type().indexExtension())) {
+

```

Comment:
This branch isn't run at all, because there's no explicit test for lessExhaustively check and we don't run SamValidator tests on anything with  a CSI index.  It would be good to add a test for this case since it's  fairly complicated.

Reply:
The discrepancy between the two BAI indexes appears to be caused by the fact that htslib uses a bin reduction [method](https://github.com/samtools/htslib/blob/bcf9bff178f81c9c1cf3a052aeb6cbe32fe5fdcc/hts.c#L1436), not implemented in htsjdk. However, both index files are functional and porting this piece of code should not be part of this PR. Thus, I kept the original BAI file.